### PR TITLE
Migrate alarm switch to alarm control panel

### LINKED ISSUE #11: Change the Alarm switch and rename it to something clearer
Hey, 

I wonder whether the `switch.biketrax_is_guarded` could be renamed to something clearer. It took me a minute the first time to realize it was actually the big button in the app to arm/disarm the alarm. Maybe something like `Arm alarm` instead of just `Alarm`? And for the entity `switch.biketrax_is_alarm_armed`?

And thinking of this, I was actually wondering whether it would be worth changing this entity from a switch to a lock or an actual alarm control panel, although I think the latter would be overkill, as there is no code nor home/night/away modes. 

What are your thoughts?

Cheers

### PR DESCRIPTION
**Pull request by vlebourl**
Closes #11 

This replaces the `Alarm` switch that controls arming the alarm with an alarm control panel entity.

image
image
image
image
image

Small caveat, the UI always shows both Arm Home and Arm Away even if only one of them is added as a supported feature. To circumvent this, I added the local bool `_is_home` to present the state `Armed Home` if it's the one that has been selected by the user. This bool is set to false when disarmed, so the next time the alarm is armed from auto guarding for instance, `Armed Away` will be shown. 

From Home Assistant, it's quite easy to set an automation that arms with `Arm Home` if the device is detected at home, so I don't think that this caveat is such a problem.

### CODE REVIEW

**Comment by vlebourl** [comment]
Trigger is synced with the stolen state:

triggered:
image

not triggered:
image

this also allows to us the `alarm_control_panel.trigger` service to both trigger the alarm and turn on the `stolen` switch.

**Comment by vlebourl** [comment]
Hey, now that version 0.2 has been out for a while, any chance you might reconsider this? I have it installed on my own prod instance through my fork, and it's working like a charm!

**Comment by basilfx** [comment]
Yes, I will definitely consider this for a 0.3.0 version, but I'm currently busy with holidays and some other projects.

**Comment by vlebourl** [comment]
No rush! Enjoye your holidays!

File: .github/pr-labeler.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+feature: ['feature/*', 'feat/*']
```
Comment (basilfx) [inline .github/pr-labeler.yml:1]: I'm not planning in using the pr-labeler (yet). Can you remove it from this PR?

File: .github/workflows/pr-labeler.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: PR Labeler
```
Comment (basilfx) [inline .github/workflows/pr-labeler.yml:1]: I'm not planning in using the pr-labeler (yet). Can you remove it from this PR?

File: custom_components/biketrax/alarm_control_panel.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+"""Support for Overkiz Alarms."""
```
Comment (basilfx) [inline custom_components/biketrax/alarm_control_panel.py:1]:
```suggestion
"""Device alarm control panel for BikeTrax devices."""
```

**Review by basilfx — CHANGES_REQUESTED** [review]

**Comment by basilfx** [comment]
I finally found some time to take a look. I think it is a good addition, but I do have some doubts about the 'arm away' and 'arm home' situation. I do understand that its because of a bug in HASS: https://github.com/nielsfaber/alarmo/issues/384 (which is the opposite of this).

If we decide to remove 'arm home' (or 'arm away', although I would rather have an additional mode called 'arm'), then we end up with a broken UI. If we do include it, the status would be reset to 'arm away' after a restart.

From a usability perspective, I think it would be OK to leave it like this, and maybe fix it in the future. In that case, I could you add a code comment to highlight the whole issue? Maybe refer to this PR for more information? The known issues in the README also needs an update.

**Comment by basilfx** [comment]
And if possible, can you try to rebase your changes on top of master? I try to keep merge commits out of feature branches, so the history of master stays 'clean'. If that doesn't work, just add the commits necessary and I'll do the rebasing afterwards.

**Comment by basilfx** [comment]
I'll probably not right away, but somewhere this week, so I can 'gather' and 'fix' 0.3.0 issues if necessary. This will be 0.4.0.

**Comment by vlebourl** [comment]
see #46.

**Pull request closed without merging**